### PR TITLE
Fix dev version when pushing to TestPyPi

### LINKED ISSUE #11: Dev version is pushed wrongly
Currently, we are using the released version for the dev. For instance, if we release the CLI with version `0.7.0` the dev should push the `0.8.0devN` but it pushes the `0.7.0.devN`.

### PR DESCRIPTION
**Pull request by raminqaf**
Closes #11

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -18,41 +18,15 @@ jobs:
         os: [ Ubuntu, MacOS, Windows ]
         python-version: [ 3.7, 3.8, 3.9 ]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+      # https://github.com/bakdata/ci-templates/blob/v1.2.0/actions/python-setup-poetry/action.yaml
```
Comment (torbsto) [inline .github/workflows/ci.yaml]: I don't see the value in this comment

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -18,41 +18,15 @@ jobs:
         os: [ Ubuntu, MacOS, Windows ]
         python-version: [ 3.7, 3.8, 3.9 ]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+      # https://github.com/bakdata/ci-templates/blob/v1.2.0/actions/python-setup-poetry/action.yaml
+      - name: Install python and set up Poetry
```
Comment (torbsto) [inline .github/workflows/ci.yaml]:
```suggestion
      - name: Install Python and set up Poetry

```

File: .github/workflows/publish-dev.yaml [hunk 3]
```diff
@@ -10,58 +10,24 @@ env:
...
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+      # https://github.com/bakdata/ci-templates/blob/v1.2.0/actions/python-setup-poetry/action.yaml
+      - name: Install python and set up Poetry and bump version with poetry
```
Comment (torbsto) [inline .github/workflows/publish-dev.yaml]:
```suggestion
      - name: Install Python and set up Poetry and bump version with Poetry
```

File: .github/workflows/publish-dev.yaml [hunk 4]
```diff
@@ -79,6 +45,7 @@ jobs:
         shell: bash
 
       - name: Build and Publish package to TestPyPI
+        if: ${{ success() }}
```
Comment (torbsto) [inline .github/workflows/publish-dev.yaml]: Why do we need this? I thought this wouldn't be executed, if a previous job failed.

Comment (raminqaf) [reply .github/workflows/publish-dev.yaml]: Okay, according to the [Docs](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions):
```
A default status check of success() is applied unless you include one of these functions
```

**Review by torbsto — APPROVED** [review]

**Pull request merged**